src/compiler/emitter.ts

```diff
@@ -1639,10 +1639,12 @@ module ts {
             }
 
             if (root) {
+                currentSourceFile = root;
                 emit(root);

```

Comment:
Why doesn't the `emit` method set `currentSourceFile`? Seems hacky to do it here.